comp/core/settings/settingsimpl/settingsimpl_test.go

```diff
@@ -131,67 +134,76 @@ func TestRuntimeSettings(t *testing.T) {
 			func(t *testing.T, comp settings.Component) {
-				responseRecorder := httptest.NewRecorder()
-
-				request := httptest.NewRequest("GET", "http://agent.host/test/", nil)
+				router := mux.NewRouter()
+				router.HandleFunc("/config/{setting}", comp.GetValue).Methods("GET")
+				ts := httptest.NewServer(router)
+				defer ts.Close()
 
-				comp.GetValue("foo", responseRecorder, request)
+				request, err := http.NewRequest("GET", ts.URL+"/config/foo", nil)
+				require.NoError(t, err)
 
-				resp := responseRecorder.Result()
-				defer resp.Body.Close()
+				resp, err := ts.Client().Do(request)
+				require.NoError(t, err)
 				body, _ := io.ReadAll(resp.Body)
+				resp.Body.Close()
 
-				assert.Equal(t, 200, responseRecorder.Code)
+				assert.Equal(t, 200, resp.StatusCode)
 				assert.Equal(t, "{\"value\":{\"Value\":\"\",\"Source\":\"\"}}", string(body))
 
-				requestWithSources := httptest.NewRequest("GET", "http://agent.host?sources=true", nil)
-				responseRecorder2 := httptest.NewRecorder()
+				requestWithSources, err := http.NewRequest("GET", ts.URL+"/config/foo?sources=true", nil)
+				require.NoError(t, err)
 
-				comp.GetValue("foo", responseRecorder2, requestWithSources)
-				resp2 := responseRecorder2.Result()
-				defer resp2.Body.Close()
-				body, _ = io.ReadAll(resp2.Body)
+				resp, err = ts.Client().Do(requestWithSources)
+				require.NoError(t, err)
+				body, _ = io.ReadAll(resp.Body)
+				resp.Body.Close()
 
-				assert.Equal(t, 200, responseRecorder2.Code)
+				assert.Equal(t, 200, resp.StatusCode)
 				assert.Equal(t, "{\"sources_value\":[{\"Source\":\"default\",\"Value\":null},{\"Source\":\"unknown\",\"Value\":null},{\"Source\":\"file\",\"Value\":null},{\"Source\":\"environment-variable\",\"Value\":null},{\"Source\":\"agent-runtime\",\"Value\":null},{\"Source\":\"local-config-process\",\"Value\":null},{\"Source\":\"remote-config\",\"Value\":null},{\"Source\":\"cli\",\"Value\":null}],\"value\":{\"Value\":\"\",\"Source\":\"\"}}", string(body))
 
-				responseRecorder3 := httptest.NewRecorder()
+				unknownSettingRequest, err := http.NewRequest("GET", ts.URL+"/config/non_existing", nil)
+				require.NoError(t, err)
 
-				comp.GetValue("non_existing", responseRecorder3, request)
-				resp3 := responseRecorder3.Result()
-				defer resp3.Body.Close()
-				body, _ = io.ReadAll(resp3.Body)
+				resp, err = ts.Client().Do(unknownSettingRequest)
+				require.NoError(t, err)
+				body, _ = io.ReadAll(resp.Body)
+				resp.Body.Close()
 
-				assert.Equal(t, 400, responseRecorder3.Code)
+				assert.Equal(t, 400, resp.StatusCode)
 				assert.Equal(t, "{\"error\":\"setting non_existing not found\"}\n", string(body))
 			},
 		},
 		{
 			"SetValue",
 			func(t *testing.T, comp settings.Component) {
-				responseRecorder := httptest.NewRecorder()
+				router := mux.NewRouter()
+				router.HandleFunc("/config/{setting}", comp.GetValue).Methods("GET")
+				router.HandleFunc("/config/{setting}", comp.SetValue).Methods("POST")
```

Comment:
Maybe we can move the router regisration outside of the testing functions. Maybe in line 239. That way we do not have to create a new router on every test iteration 😄